crates/cli/src/codegen/mod.rs

```diff
@@ -106,43 +111,88 @@ impl BytecodeMetadata {
 }
 
 /// Code Generation.
-pub(crate) struct Generator {
-    /// Codegen type to use.
-    pub ty: CodeGenType,
-    /// JS runtime config.
-    pub js_runtime_config: JsConfig,
+#[derive(Default, Clone)]
+pub struct Generator {
     /// Plugin to use.
-    pub plugin: Plugin,
-    /// JavaScript function exports.
-    function_exports: Exports,
+    pub(crate) plugin: plugin::Plugin,
+    /// What kind of linking to use when generating.
+    pub(crate) linking: LinkingKind,
     /// Whether to embed the compressed JS source in the generated module.
-    pub source_compression: bool,
+    pub(crate) source_compression: bool,
     /// WIT options for code generation.
-    pub wit_opts: WitOptions,
+    pub(crate) wit_opts: wit::WitOptions,
+    /// JavaScript function exports.
+    pub(crate) function_exports: exports::Exports,
+    /// The kind of plugin a generator will link.
+    plugin_kind: plugin::PluginKind,
```

Comment:
This will never be exposed directly to customers. I know you mentioned removing the `PluginKind` entirely from the `codegen` crate but storing boolean flags for what we're linking against is actually more complex. I think this solution of abstracting the `PluginKind` behind boolean setters helps create the interface you're looking for while slightly reducing the implementation complexity (mostly around error handling and checking if each boolean is set).

Reply:
I'm totally okay with having `PluginKind` in the generator if it's not exposed in the public API used by most consumers 👍